packages/deployments/contracts/contracts/core/connext/facets/TokenFacet.sol

```diff
@@ -359,6 +360,16 @@ contract TokenFacet is BaseConnextFacet {
     // Sanity check: already approval
     if (!s.approvedAssets[_key]) revert TokenFacet__removeAssetId_notAdded();
 
+    TokenId memory adoptedCanonical = s.adoptedToCanonical[_adoptedAssetId];
+    TokenId memory representationCanonical = s.adoptedToCanonical[_adoptedAssetId];
+
+    // Sanity check: consistent set of params
+    if (
+      AssetLogic.calculateCanonicalHash(adoptedCanonical.id, adoptedCanonical.domain) != _key ||
+      adoptedCanonical.id != representationCanonical.id ||
+      adoptedCanonical.domain != representationCanonical.domain
+    ) revert TokenFacet__removeAssetId_invalidParams();
```

Comment:
This seems a bit more readable:

```suggestion
    // Sanity check: consistent set of params
    if (
      s.canonicalToAdopted[_key] != _adopted ||
      s.canonicalToRepresentation[_key] != _local
    ) revert TokenFacet__removeAssetId_invalidParams();
```

well want to make sure there are tests for removing assets on the canonical domain. the setup asset tests just got fixed in #2455 but `removeAsset` was not updated. just make that PR a blocker to this one and merge it in.

Reply:
pushed the suggested changes. Not sure how to make the other pr a blocker and merge it in :(